test/test_sparse_csr.py

```diff
@@ -2798,6 +2798,31 @@ def run_test(shape, nnz, index_type):
             run_test(shape, max(shape), index_dtype)
             run_test(shape, shape[0] * shape[1], index_dtype)
 
+    @skipMeta
+    @dtypes(*all_types_and_complex_and(torch.half, torch.bfloat16))
+    def test_sum_dim_reduce(self, device, dtype):
+        def run_test(shape, nnz, index_type):
+            sparse = self.genSparseCSRTensor(shape, nnz, dtype=dtype, device=device, index_dtype=index_dtype)
+            dense = sparse.to_dense()
+            for dim in (0, 1, (0, )):
+                dense_sum = dense.sum(dim=dim)
+                sparse_sum = sparse.sum(dim=dim, keepdim=True)
+                is_integral = dtype in integral_types()
+                self.assertEqual(sparse_sum.to_dense().view(dense_sum.shape)
+                                 if not is_integral else sparse_sum.to_dense().to(torch.int64).view(dense_sum.shape), dense_sum)
```

Comment:
We should have the following invariant holding:
```
sparse.sum(dim=dim, keepdim=True).to_dense() == sparse.to_dense().sum(dim=dim, keepdim=True)
```
So:
```suggestion
                dense_sum = dense.sum(dim=dim, keepdim=True)
                sparse_sum = sparse.sum(dim=dim, keepdim=True)
                self.assertEqual(sparse_sum, dense_sum)
```
(`assertEqual` with handle the conversion of `sparse_sum` to a strided tensor).